src/services/AuthorizationService.ts

```diff
@@ -1,36 +1,35 @@
- *--------------------------------------------------------------------------------------------*/
-import { ClientRequestContext } from "@bentley/bentleyjs-core";
-import { BrowserAuthorizationClient, BrowserAuthorizationClientConfiguration } from "@bentley/frontend-authorization-client";
-
-export class AuthorizationService {
-  private accessToken: string | undefined = undefined;
-
-  private async authorize(): Promise<void> {
-    if (!process.env.IMJS_AUTH_CLIENT_ID || !process.env.IMJS_AUTH_REDIRECT_URL || !process.env.IMJS_AUTH_CLIENT_SCOPES)
-      throw new Error("Missing configuration. Keys IMJS_AUTH_CLIENT_ID, IMJS_AUTH_REDIRECT_URL, IMJS_AUTH_CLIENT_SCOPES must have values. Please provide them in the .env file.");
-
-    const oidcConfiguration: BrowserAuthorizationClientConfiguration = {
-      authority: process.env.IMJS_AUTH_CLIENT_AUTHORITY,
-      clientId: process.env.IMJS_AUTH_CLIENT_ID,
-      redirectUri: process.env.IMJS_AUTH_REDIRECT_URL,
-      postSignoutRedirectUri: undefined,
-      scope: process.env.IMJS_AUTH_CLIENT_SCOPES,
-      responseType: "code",
-    };
-
-    const oidcClient = new BrowserAuthorizationClient(oidcConfiguration);
-    oidcClient.onUserStateChanged.addListener((token) => this.accessToken = token?.toTokenString() ?? "");
-    await oidcClient.signIn(new ClientRequestContext());
-  }
-
-  public async getAccessToken(): Promise<string> {
-    if (this.accessToken)
-      return this.accessToken;
-
-    await this.authorize();
-    return this.accessToken!;
-  }
-}
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { BrowserAuthorizationClient, BrowserAuthorizationClientConfiguration } from "@itwin/browser-authorization";
+
+export class AuthorizationService {
+  private _authorizationClient: BrowserAuthorizationClient | undefined;
+
+  private async createAuthorizationClient(): Promise<BrowserAuthorizationClient> {
+    if (!process.env.IMJS_AUTH_CLIENT_ID || !process.env.IMJS_AUTH_REDIRECT_URL || !process.env.IMJS_AUTH_CLIENT_SCOPES)
+      throw new Error("Missing configuration. Keys IMJS_AUTH_CLIENT_ID, IMJS_AUTH_REDIRECT_URL, IMJS_AUTH_CLIENT_SCOPES must have values. Please provide them in the .env file.");
+
+    const authorizationClientConfiguration: BrowserAuthorizationClientConfiguration = {
+      authority: process.env.IMJS_AUTH_CLIENT_AUTHORITY,
+      clientId: process.env.IMJS_AUTH_CLIENT_ID,
+      redirectUri: process.env.IMJS_AUTH_REDIRECT_URL,
+      postSignoutRedirectUri: undefined,
+      scope: process.env.IMJS_AUTH_CLIENT_SCOPES,
+      responseType: "code",
+    };
+
+    const client = new BrowserAuthorizationClient(authorizationClientConfiguration);

```

Comment:
You could always try a silentSignin directly after creating the client. That way it prevents a user from getting prompted if they're already logged in within the current browser session.

Reply:
It looks like the implementation already attempts a silent login: [`signIn`](https://github.com/iTwin/auth-clients/blob/31bf6894a9a51073d873c818d99a76c60a11a427/packages/browser/src/Client.ts#L167) -> [`signInRedirect`](https://github.com/iTwin/auth-clients/blob/31bf6894a9a51073d873c818d99a76c60a11a427/packages/browser/src/Client.ts#L178) -> `nonInteractiveSignIn` which is also called by [`signInSilent`](https://github.com/iTwin/auth-clients/blob/31bf6894a9a51073d873c818d99a76c60a11a427/packages/browser/src/Client.ts#L215)